### PR TITLE
fix: repeated cleaning dom (#3665)

### PR DESCRIPTION
**Pull request by zxyao145**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
#3665

### 📝 Changelog

**break changes**
When the URI changes, the Dialog box created through `ModalServce` or `ConfirmService`, and the Drawer created  through 
`DrawerService` will be forcibly cleared.

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     fix: repeated cleaning dom      |
| 🇨🇳 Chinese |      修复重复移除DOM的问题   |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/core/JsInterop/modules/dom/manipulationHelper.ts
Comment (ElderJames) [inline components/core/JsInterop/modules/dom/manipulationHelper.ts:1]: Please revert the changes unrelated to this PR.

File: components/drawer/DrawerContainer.razor.cs [hunk 1]
```diff
@@ -11,11 +11,20 @@ public partial class DrawerContainer
...
+        [Inject]
+        private NavigationManager NavigationManager { get; set; }
+
         protected override void OnInitialized()
         {
             DrawerService.OnCloseEvent += DrawerService_OnClose;
             DrawerService.OnOpenEvent += DrawerService_OnCreate;
             DrawerService.OnUpdateEvent += DrawerService_OnUpdateEvent;
+
+            NavigationManager.LocationChanged += (sender, e) =>
```
Comment (ElderJames) [inline components/drawer/DrawerContainer.razor.cs:23]: The event handler should be remove when component is disposed.

File: components/core/JsInterop/modules/dom/manipulationHelper.ts
Comment (zxyao145) [reply components/core/JsInterop/modules/dom/manipulationHelper.ts:1]: @ElderJames Each developer has their own formatting style, and I believe that a lint file is more important than fixing it. Could you add an eslint file?

File: components/modal/modalDialog/Modal.razor.cs [hunk 2]
```diff
@@ -377,6 +382,29 @@ private async Task OnAfterDialogShow()
                     await ModalRef.OnOpen();
                 }
             }
+
+            if (_firstShow)
+            {
+                _firstShow = false;
+                NavigationManager.LocationChanged += (sender, e) =>
```
Comment (ElderJames) [inline components/modal/modalDialog/Modal.razor.cs:389]: Can this be removed when the instance is disposed?

**Review by ElderJames — APPROVED** [review]
